### PR TITLE
added fragment

### PR DESCRIPTION
**Pull request by LuitelSamikshya**
This PR adds fragment support to stop url fragments being stripped.

### CODE REVIEW

**Review by Mzack9999 — CHANGES_REQUESTED** [review]
- [ ] Remove code related to https://github.com/projectdiscovery/urlutil/pull/2
- [ ] Review https://github.com/projectdiscovery/urlutil/pull/2, merge it into `dev`, then merge `dev` into this branch
- [ ] Please add a little test case related to Fragment parsing in `urlutil_test.go`

Notes: depends on https://github.com/projectdiscovery/urlutil/pull/2

File: urlutil.go [hunk 1]
```diff
@@ -84,14 +105,23 @@ func ParseWithScheme(u string) (*URL, error) {
 	if ruri := U.RequestURI(); ruri != "/" {
 		requri = ruri
 	}
+	if origReqURI != "" && origReqURI != "/" {
+		requri = origReqURI
+	}
 
+	//fragment
```
[the remark this answers is not in the document]
Comment (LuitelSamikshya) [reply urlutil.go:112]: Done

File: urlutil.go [hunk 2]
```diff
@@ -51,9 +58,23 @@ func ParseWithScheme(u string) (*URL, error) {
 	// prepend default scheme if absent to increase parsing capabilities
 	u = PreprendDefaultScheme(u)
 
+	var origReqURI string
 	U, err := url.Parse(u)
 	if err != nil {
-		return nil, err
+		// try to reparse without the request path
```
[the remark this answers is not in the document]
Comment (LuitelSamikshya) [reply urlutil.go]: Done

File: urlutil.go [hunk 3]
```diff
@@ -40,6 +41,12 @@ func (u URL) String() string {
 		fullURL.WriteString(":" + u.Port)
 	}
 	fullURL.WriteString(u.RequestURI)
+	if u.Fragment != "" && u.RequestURI == "" {
```
[the remark this answers is not in the document]
Comment (LuitelSamikshya) [reply urlutil.go]: Done

**Review by Mzack9999 — APPROVED** [review]

**Pull request merged**